### PR TITLE
Doxygen updates

### PR DESCRIPTION
**Pull request by madebr**
- exclude the correct include files from doxygen
- add `\brief` to (almost) all include headers such that it is displayed in the files toc.
- document macro's of SDL_endian.h

### CODE REVIEW

File: include/SDL3/SDL_revision.h [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+freely, subject to the following restrictions:
+
+1. The origin of this software must not be misrepresented; you must not
+claim that you wrote the original software. If you use this software
+in a product, an acknowledgment in the product documentation would be
+appreciated but is not required.
+2. Altered source versions must be plainly marked as such, and must not be
+misrepresented as being the original software.
+3. This notice may not be removed or altered from any source distribution.
+*/
```
Comment (slouken) [inline include/SDL3/SDL_revision.h:20]: Spacing is wrong here, FYI

File: include/build_config/SDL_revision.h.cmake [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+freely, subject to the following restrictions:
+
+1. The origin of this software must not be misrepresented; you must not
+claim that you wrote the original software. If you use this software
+in a product, an acknowledgment in the product documentation would be
+appreciated but is not required.
+2. Altered source versions must be plainly marked as such, and must not be
+misrepresented as being the original software.
+3. This notice may not be removed or altered from any source distribution.
+*/
```
Comment (slouken) [inline include/build_config/SDL_revision.h.cmake:20]: spacing is wrong here as well. Did you copy SDL_copying.h in?

Comment (madebr) [reply include/build_config/SDL_revision.h.cmake:20]: I don't know where I got it from exactly. It looks like my ide (clion) removes the indentation.

**Comment by slouken** [comment]
Looks good, thanks!

**Pull request merged**